dirsrvtests/tests/suites/replication/regression_m2_test.py

```diff
@@ -1144,6 +1148,129 @@ def test_bulk_import(preserve_topo_m2):
+    agmt = Agreements(S1).list()[0]
+    agmt.begin_reinit()
+    (done, error) = agmt.wait_reinit()
+    assert done is True
+    assert error is False
+
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+    repl.wait_for_replication(S1, S2)
+    repl.wait_for_replication(S2, S1)
+
+
+def test_rid_starting_with_0(topo_m2, request):
+    """Check that replication monitoring works if replica
+       id starts with 0
+
+    :id: ed0176e6-0bf7-11f0-9846-482ae39447e5
+    :setup: 2 Supplier Instances
+    :steps:
+        1. Initialize replication to ensure that init status is set
+        2. Check that monitoring status does not contains 'Unavailable'
+        3. Change replica ids to 001 and 002
+        4. Initialize replication to ensure that init status is set
+        5. Check that monitoring status does not contains 'Unavailable'
+        6. Restore the replica ids to 1 and 2
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+    """
+    S1 = topo_m2.ms["supplier1"]
+    S2 = topo_m2.ms["supplier2"]
+    replicas = [ Replicas(inst).get(DEFAULT_SUFFIX) for inst in topo_m2 ]
+
+    # Reinit replication (to ensure that init status is set)
+    reinit_replica(S1, S2)
+
+    # Get replication monitoring results
+    check_monitoring_status(S1)
+
+    # Change replica id
+    for replica,rid in zip(replicas, ['010', '020']):
+        replica.replace('nsDS5ReplicaId', rid)
+
+    # Restore replica id in finalizer
+    def fin():
+        for replica,rid in zip(replicas, ['1', '2']):
+            replica.replace('nsDS5ReplicaId', rid)
+        reinit_replica(S1, S2)
+
+    request.addfinalizer(fin)
+    # Reinit replication
+    reinit_replica(S1, S2)
+
+    # Get replication monitoring results
+    check_monitoring_status(S1)
+
+
+def test_normalized_rid_dict():
+    """Check that lib389.replica NormalizedRidDict class behaves as expected
+
+    :id: 0f88a29c-0fcd-11f0-b5df-482ae39447e5
+    :setup: None
+    :steps:
+        1. Initialize a NormalizedRidDict
+        2. Check that normalization do something
+        3. Check that key stored in NormalizedRidDict are normalized
+        4. Check that normalized and non normalized keys have the same value
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+    """
+
+    sd = { '1': 'v1', '020': 'v2' }
+    nsd = { NormalizedRidDict.normalize_rid(key): val for key,val in sd.items() }
+    nkeys = list(nsd.keys())
+
+    # Initialize a NormalizedRidDict
+    nrd = NormalizedRidDict()
+    for key,val in sd.items():
+        nrd[key] = val
+
+    # Check that normalization do something
+    assert nkeys != [ sd.keys() ]
+
+    # Check that key stored in NormalizedRidDict are normalized
+    for key in nrd.keys():
+        assert key in nkeys
+
+    # Check that normalized and non normalized keys have the same value
+    for keyl in sd.items():
```

Comment:
probably, it should be `for key, val in sd.items():`

Reply:
Ok. 
I  do not understand how the test could PASS with such an error ...
dirsrvtests/tests/suites/replication/regression_m2_test.py::test_normalized_rid_dict PASSED [ 93%]

Finally I understand: there was two error, keyl instead of key and the missing val 
so the test was using the same values for key and val in each loop ...